src/Microsoft.AspNetCore.Mvc.Razor/Internal/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -56,7 +57,6 @@ public IEnumerable<TagHelperDescriptor> CreateDescriptors(string assemblyName)
                     StringComparison.Ordinal));
 
             var tagHelperDescriptors = viewComponentDescriptors
-                .Where(descriptor => !descriptor.Parameters.Any(parameter => parameter.ParameterType.GetTypeInfo().IsGenericType))
```

Comment:
So if it is invoked the usual way (@Component.Invoke) it provides a good validation message. But the same does not happen with VCTH because that code path doesn't run. Even if we don't protect against it we should make it not show up in intellisense right?